raiden/network/proxies/token_network.py

```diff
@@ -1085,3 +1055,133 @@ def _check_for_outdated_channel(
+
+        if participant_details.our_details.deposit > deposit_amount:
+            raise RaidenUnrecoverableError('Deposit amount decreased')
+
+        channel_state = self._get_channel_state(
+            participant1=self.node_address,
+            participant2=participant2,
+            channel_identifier=channel_identifier,
+        )
+        # Check if deposit is being mode on a nonexistent channel
+        if channel_state in (ChannelState.NONEXISTENT, ChannelState.REMOVED):
+            raise RaidenUnrecoverableError(
+                f'Channel between participant {participant1} '
+                f'and {participant2} does not exist',
+            )
+        # Deposit was prohibited because the channel is settled
+        elif channel_state == ChannelState.SETTLED:
+            raise RaidenUnrecoverableError(
+                'Channel is settled',
+            )
+        # Deposit was prohibited because the channel is closed
+        elif channel_state == ChannelState.CLOSED:
+            raise RaidenRecoverableError(
+                'Channel is not in an open state',
+            )
+
+    def _check_channel_state_for_withdraw(
+            self,
+            participant1,
+            participant2,
+            channel_identifier,
+            withdraw_amount,
+    ):
+        participant_details = self.detail_participants(
+            participant1,
+            participant2,
+            channel_identifier,
+        )
+
+        if participant_details.our_details.withdrawn > withdraw_amount:
+            raise WithdrawMismatch('Withdraw amount decreased')
+
+        channel_state = self._get_channel_state(
+            participant1=participant1,
+            participant2=participant2,
+            channel_identifier=channel_identifier,
+        )
+
+        if channel_state in (ChannelState.NONEXISTENT, ChannelState.REMOVED):
+            raise RaidenUnrecoverableError(
+                f'Channel between participant {participant1} '
+                f'and {participant2} does not exist',
+            )
+        # Deposit was prohibited because the channel is settled
+        elif channel_state == ChannelState.SETTLED:
+            raise RaidenUnrecoverableError(
+                'Channel is not in an settled state',
+            )
+        # Deposit was prohibited because the channel is closed
+        elif channel_state == ChannelState.CLOSED:
+            raise RaidenRecoverableError(
+                'Channel is not in an open state',
+            )
+
+    def _check_channel_state_for_settle(self, participant1, participant2, channel_identifier):
+        channel_data = self.detail_channel(participant1, participant2, channel_identifier)
+        if channel_data.state == ChannelState.SETTLED:
+            raise RaidenRecoverableError(
+                'Channel is not in a closed state. It cannot be settled',
+            )
+        elif channel_data.state == ChannelState.REMOVED:
+            raise RaidenUnrecoverableError(
```

Comment:
This is actually not unrecoverable, if there are no pending unlocks the channel will transition directly to the removed state, so it can be a race just like the settled above.